aptos-node/src/lib.rs

```diff
@@ -112,41 +197,40 @@ pub fn start(config: NodeConfig, log_file: Option<PathBuf>) {
 
 const EPOCH_LENGTH_SECS: u64 = 60;
 
+// Loads the test environment, and loads the previous run if it already exists
 pub fn load_test_environment<R>(
     config_path: Option<PathBuf>,
+    node_dir: PathBuf,
     random_ports: bool,
     lazy: bool,
     genesis_modules: Vec<Vec<u8>>,
     rng: R,
 ) where
     R: ::rand::RngCore + ::rand::CryptoRng,
 {
-    let config_temp_path = aptos_temppath::TempPath::new();
-
-    let (try_load, config_path) = if let Some(config_path) = config_path {
-        (
-            config_path.join("0").join("node.yaml").exists(),
-            config_path,
-        )
-    } else {
-        (false, config_temp_path.as_ref().to_path_buf())
-    };
-
     std::fs::DirBuilder::new()
         .recursive(true)
-        .create(&config_path)
+        .create(&node_dir)
         .unwrap();
 
-    let config_path = config_path.canonicalize().unwrap();
-
-    let validator_config_path = config_path.join("0").join("node.yaml");
-    let aptos_root_key_path = config_path.join("mint.key");
-
-    let config = if try_load {
-        NodeConfig::load(&validator_config_path).expect("Unable to load config:")
+    // The directory that will hold all the data about the network
+    let node_dir = node_dir.canonicalize().unwrap();
+
+    let validator_config_path = node_dir.join("0").join("node.yaml");
+    let aptos_root_key_path = node_dir.join("mint.key");
+
+    let config = if let Some(config_path) = config_path {
+        // Copy the config into the node directory
+        let config = NodeConfig::load(&config_path).expect("Unable to load config:");
+        std::fs::copy(config_path, validator_config_path.as_path())
+            .expect("Should be able to copy config");
+        config
+    } else if let Ok(config) = NodeConfig::load(&validator_config_path) {
+        // If there's an existing config, use that
+        config
```

Comment:
This seems like a potential point of confusion, since there will now be two configs. Can we just reference the existing one instead of copying it?